frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/AuthenticationAuthCodeHandler.java

```diff
@@ -0,0 +1,97 @@
+import uk.gov.di.authentication.shared.state.UserContext;
+
+import java.net.URI;
+
+import static uk.gov.di.authentication.shared.helpers.ApiGatewayResponseHelper.generateApiGatewayProxyErrorResponse;
+import static uk.gov.di.authentication.shared.helpers.ApiGatewayResponseHelper.generateApiGatewayProxyResponse;
+import static uk.gov.di.authentication.shared.helpers.LogLineHelper.LogFieldName.AWS_REQUEST_ID;
+import static uk.gov.di.authentication.shared.helpers.LogLineHelper.attachLogFieldToLogs;
+
+public class AuthenticationAuthCodeHandler extends BaseFrontendHandler<AuthCodeRequest> {
+
+    private static final Logger LOG = LogManager.getLogger(AuthenticationAuthCodeHandler.class);
+
+    private final DynamoAuthCodeService dynamoAuthCodeService;
+
+    public AuthenticationAuthCodeHandler(
+            DynamoAuthCodeService dynamoAuthCodeService,
+            ConfigurationService configurationService,
+            SessionService sessionService,
+            ClientSessionService clientSessionService,
+            ClientService clientService,
+            AuthenticationService authenticationService) {
+        super(
+                AuthCodeRequest.class,
+                configurationService,
+                sessionService,
+                clientSessionService,
+                clientService,
+                authenticationService);
+        this.dynamoAuthCodeService = dynamoAuthCodeService;
+    }
+
+    public AuthenticationAuthCodeHandler(ConfigurationService configurationService) {
+        super(AuthCodeRequest.class, configurationService);
+        this.dynamoAuthCodeService = new DynamoAuthCodeService(configurationService);
+    }
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequestWithUserContext(
+            APIGatewayProxyRequestEvent input,
+            Context context,
+            AuthCodeRequest request,
+            UserContext userContext) {
+        attachLogFieldToLogs(AWS_REQUEST_ID, context.getAwsRequestId());
+        if (configurationService.isAuthOrchSplitEnabled()) {
+            try {
+                var authCodeRequest =
+                        objectMapper.readValue(input.getBody(), AuthCodeRequest.class);
+                var userProfile =
+                        authenticationService.getUserProfileByEmailMaybe(request.getEmail());
+                if (userProfile.isEmpty()) {
+                    LOG.info(
+                            "Error message: Email from session does not have a user profile required to extract Subject ID");
+                    return generateApiGatewayProxyErrorResponse(400, ErrorResponse.ERROR_1049);
+                }
+
+                var authorisationCode = new AuthorizationCode();
+                dynamoAuthCodeService.saveAuthCode(
+                        userProfile.get().getSubjectID(),
+                        authorisationCode.getValue(),
+                        authCodeRequest.getRequestedScopeClaims(),
```

Comment:
I think you should be able to get this off the existing `request` being passed as an argument